lib/rails_multitenant/middleware.rb

```diff
@@ -0,0 +1,19 @@
+module RailsMultitenant
+  class Middleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      dup._call(env)
```

Comment:
Why do we need the `dup` + `_call`? Seems like we can simplify this and get rid of tracking the response in an instance variable.

Reply:
Misunderstanding around thread-safety here, re-read and updated!